lib/std/crypto/phc_encoding.zig

```diff
@@ -0,0 +1,391 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2021 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+
+// https://github.com/P-H-C/phc-string-format
+
+const std = @import("std");
+const fmt = std.fmt;
+const io = std.io;
+const mem = std.mem;
+const meta = std.meta;
+
+const fields_delimiter = "$";
+const version_param_name = "v";
+const params_delimiter = ",";
+const kv_delimiter = "=";
+
+pub const Error = std.crypto.errors.EncodingError || error{NoSpaceLeft};
+
+const B64Decoder = std.base64.standard_no_pad.Decoder;
+const B64Encoder = std.base64.standard_no_pad.Encoder;
+
+/// A wrapped binary value whose maximum size is `max_len`.
+///
+/// This type must be used whenever a binary value is encoded in a PHC-formatted string.
+/// This includes `salt`, `hash`, and any other binary parameters such as keys.
+///
+/// Once initialized, the actual value can be read with the `unwrap()` function.
+pub fn BinValue(comptime max_len: usize) type {
```

Comment:
This is the same as https://github.com/ziglang/zig/pull/9134 albeit with the base64 encoding functions.

Maybe use `slice()`/`constSlice()` instead of `unwrap()` for consistency?